www/js/lib/uiUtil.js

```diff
@@ -204,6 +216,16 @@ function systemAlert (message, label, isConfirm, declineConfirmLabel, approveCon
             document.getElementById('modalLabel').innerHTML = label;
             // Using innerHTML to set the message to allow HTML formatting
             document.getElementById('modalText').innerHTML = message;
+            // Set archive metadata if provided. Avoid using innerHTML since data can be spoofed.
+            document.getElementById('modal-archive-metadata-container').style.display = archiveMetadata ? 'block' : 'none';
+            if (archiveMetadata) {
+                document.getElementById('confirm-archive-name').textContent = metadataLabels.name + (archiveMetadata.name || '-');
+                document.getElementById('confirm-archive-creator').textContent = metadataLabels.creator + (archiveMetadata.creator || '-');
+                document.getElementById('confirm-archive-publisher').textContent = metadataLabels.publisher + (archiveMetadata.publisher || '-');
+                document.getElementById('confirm-archive-scraper').textContent = metadataLabels.scraper + (archiveMetadata.scraper || '-');
+            }
+            document.getElementById('modal-archive-metadata-warning').textContent = metadataWarningLabel;
+            document.getElementById('modal-archive-metadata-warning').style.display = archiveMetadata ? 'block' : 'none';
```

Comment:
Ditto, move all this into your own pre-formatting function. You can either gather the translated variables you need and then simply insert them into HTML, or you can create the element with `var securityModal = document.createElement(.....)`, populate it with DOM methods like here, and then simply grab the HTML `var securityModalHTML = sucurityModal.outerHTML;` and send the HTML to `uiUtil.systemAlert`.